encryptme_stats/metrics.py

```diff
@@ -622,22 +650,100 @@ def _get_ipsec_session_stats():
         return []
 
 
+def _get_wireguard_stats():
+    """Gather Wireguard statistics."""
+    try:
+        interfaces = subprocess_out(["wg", "show", "interfaces"])
+        if not isinstance(interfaces, list):
+            return
+        if interfaces and interfaces[-1] == '':
+            interfaces.pop()
+        info = []
+        for interface in interfaces:
+            peers = subprocess_out(["wg", "show", interface, "dump"])
+            if not isinstance(peers, list):
+                continue
+            if peers and peers[-1] == '':
+                peers.pop()
+            # First element in the list contains interface's data
+            # Subsequent elements contain peer's data
+            peers.pop(0)
+            now_epoch = int(datetime.utcnow().timestamp())
+            for peer in peers:
+                _, _, _, private_ip, last_handshake, bytes_up, bytes_down, _ = peer.split('\t')
+                last_handshake = int(last_handshake)
+                if (now_epoch - 120) > last_handshake:
+                    continue
+                info.append({
+                    'stats_type': 'vpn_session',
+                    'vpn_session': {
+                        'public_id': '',
+                        'private_ip': private_ip.split('/')[0],
+                        'last_handshake': last_handshake,
+                        'logged_at': now_epoch,
+                        'bytes_up': bytes_up,
+                        'bytes_down': bytes_down,
+                        'protocol': 'wireguard',
+                    }
+                })
+    except Exception as exc:
+        logging.debug("Error gathering wireguard bandwwith: %s", exc)
+        return []
+
+    return info
+
+
 def vpn_session():
     """
     Gather per-connection stats.
 
     :return: list of dictionaries with connections statistics
     """
-    empty = []
     try:
         openvpn_stat = _get_openvpn_session_stats()
         ipsec_stat = _get_ipsec_session_stats()
+        wireguard_stat = _get_wireguard_stats()
 
-        result = openvpn_stat + ipsec_stat
-        if len(result) == 0:
-            return empty
-
-        return result
+        return (openvpn_stat + ipsec_stat + wireguard_stat)
```

Comment:
please return empty list if there is no stats. that prevents us to spam stats server with empty payloads